src/kbc_modules/cc_kbc/mod.rs

```diff
@@ -215,23 +220,25 @@ impl Kbc {
         bail!("Request KBS resource: Attested but KBS still return Unauthorized")
     }
 
-    /// Convert a [`ResourceUri`] to the address of kind cc-kbc.
-    /// This function will **ignore** the kbs address the kid carries,
-    /// instead overwrite with the kbs_uri the [`Kbc`] carries.
-    /// Related issue: <https://github.com/confidential-containers/attestation-agent/issues/130>
-    pub fn resource_web_addr(&self, kid: &ResourceUri) -> Result<String> {
-        if self.kbs_uri != kid.kbs_addr {
-            warn!(
-                "The KBS address contained in the URI is not the same as the KBC's, use the KBC's.",
-            )
+    /// Convert a [`ResourceUri`] to a KBS URL.
+    pub fn resource_to_kbs_uri(&self, resource: &ResourceUri) -> Result<String> {
+        let kbs_host = self
+            .kbs_uri
+            .host_str()
+            .ok_or_else(|| anyhow!("Invalid URL: {}", self.kbs_uri))?;
+        if kbs_host != resource.kbs_addr {
+            bail!(
```

Comment:
Currently, we cannot directly throw an error when the two are different, because the kbs address in the resource URI we currently use is actually omitted——It is a null value, which means that the resource URI entered by the current caller of AA for AA is like this: `kbs:///repo/type/tag`.

Reply:
When do we use `cc_kbc` without a KBS address?

Reply:
When decrypting image annotation, the `kid` field which in Resource URI format included in the annotation is like this: `kbs:///a/b/c`.

Reply:
I see. Would the following check:

```rust
if resource.kbs_addr.len() > 0 && kbs_host != resource.kbs_addr
```

make more sense? let me give it a go.

Reply:
> I see. Would the following check:
> 
> ```rust
> if resource.kbs_addr.len() > 0 && kbs_host != resource.kbs_addr
> ```
> 
> make more sense? let me give it a go.

I think this would be OK. Thanks!
Let me explain in detail the use of URIs in our current implementation:
AA now has two APIs, one for get resource and the other for decrypt annotation. The call inputs for these two ports are different:

For the `get resource` API, the caller is `image-rs`, which obtains the `kbc-name` and `kbs-uri` (format is like `http://1.2.3.4:8080/) from the `aa_kbc_params` (format is like `cc_kbc::http://1.2.3.4:8080/) passed in from `kata-agent` config, and obtains the Resource URI (format is like `[kbs:///a/b/c`](kbs:///a/b/c%60)) from the image validation policy file. Then `image-rs` get `resource_path` (format is like `/a/b/c`) from Resource URI, and pass `resource_path` to AA together with the `kbc-name` and `kbs uri` to call AA `get resource` interface.

For the `decrypt annotation` API, the caller is `ocicrypt-rs`. Similarly, `ocicrypt-rs` obtains the `kbc-name` and `kbs-uri `from `aa-kbc-params`, but `ocicrypt-rs` does not resolve the Resource URI contained in the annotation (format is `[kbs:///a/b/c`](kbs:///a/b/c%60)) . Instead, the annotation is directly passed to AA along with the `kbc-name` and `kbs-uri` to call AA `decrypt annotation` interface.

This difference results in different processing logic for Resource URI between the two interface of AA, specifically:

The get resource port obtains the `kbs-uri` and `resource-path` in the input parameters, initializes the KBC with the `kbs uri`, recombines the `kbs uri` and `resource path` into a REAL Resource URI (Now format is like `[kbs://1.2.3.4:8080/a/b/c`](kbs://1.2.3.4:8080/a/b/c%60)), and converts the RESL Resource URI into a real URL (format is `[http://1.2.3.4:8080/kbs/v0/a/b/c`](http://1.2.3.4:8080/kbs/v0/a/b/c%60)).

The decrypt annotation port obtains the `kbs uri `in the input parameters, and initialize the KBC using the `kbs uri`. It then parses the Resource URI in the annotation (format is `[kbs:///a/b/c`](kbs:///a/b/c%60)）And try to convert this Resource URI into a real URL (format is `[http://1.2.3.4:8080/kbs/v0/a/b/c`](http://1.2.3.4:8080/kbs/v0/a/b/c%60))

Therefore, if we throw an error here, there is no problem with `get resource` interface, but there will be an error with `decrypt annotation`.

Reply:
@sameo Sorry for the text format confusion, I have fixed these text confusion, you can refresh the page:-)

Reply:
There is another point to note, although I said above that `image-rs` and `ocictrypt-rs` get `kbs-uri` in format of `http://1.2.3.4:8080` from `aa_kbc_params`. However, due to some historical reasons, in the current code, `eaa_kbc` and `sev_kbc` require `kbs-uri` in `aa_kbc_params` must not contain a scheme prefix (`http://`)

Reply:
Thank you for that explanation. Is there a reason why `ocicrypt-rs` does not disassemble the resource URI?

Reply:
cc @Xynnn007

Reply:
It seems that I have some problems with text conversion. I fixed them. You can refresh the page again :-)

Reply:
@mkulke And the concept of ocicrypt "Key Provider" is defined here: https://github.com/containers/ocicrypt/blob/main/docs/keyprovider.md

Reply:
@jialez0 I added a few fixes. Let me know if that looks better and work for you.